core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -719,7 +721,27 @@ object SparkSubmit {
       printWarning("Subclasses of scala.App may not work correctly. Use a main() method instead.")
     }
 
-    val mainMethod = mainClass.getMethod("main", new Array[String](0).getClass)
+    val isSparkApp = mainClass.getMethods().filter(_.getName() == "sparkMain").length > 0
+
+    val childSparkConf = sysProps.filter( p => p._1.startsWith("spark.")).toMap
+    val childSysProps = sys.env
+    if (isSparkApp && threadEnabled) {
```

Comment:
Why check for `threadEnabled` here? If it's a `SparkApp` whether it's being run by a spark launcher thread should be irrelevant.